endpoints/api/robot.py

```diff
@@ -121,6 +122,12 @@ def put(self, robot_shortname):
         """
         parent = get_authenticated_user()
         create_data = request.get_json(silent=True) or {}
+
+        if model.robot_account_exists(robot_shortname, parent, user=True) is not None:
```

Comment:
Is it better to catch the duplicate name error returned by `create_user_robot` and re-raise the request_error? That way we don't add any additional API requests. Same for the org endpoint.